libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -538,6 +539,42 @@ void AP_DDS_Client::on_request(uxrSession* uxr_session, uxrObjectId object_id, u
         }
         break;
     }
+    case services[to_underlying(ServiceIndex::MODE_SWITCH)].rep_id: {
+        uint8_t mode;
+        const bool deserialize_success = ucdr_deserialize_uint8_t(ub,&mode);
+        if (deserialize_success == false) {
+            break;
+        }
+        bool status = AP::vehicle()->set_mode(mode, ModeReason::DDS_COMMAND);
+        uint8_t curr_mode = AP::vehicle()->get_mode();
+
+        const uxrObjectId replier_id = {
+            .id = services[to_underlying(ServiceIndex::MODE_SWITCH)].rep_id,
+            .type = UXR_REPLIER_ID
+        };
+
+        //Todo : Fix the size-handling of services with the help of the functions autogenerated via Micro-XRCE-DDS Gen
+        uint8_t reply_buffer[8] {};
+        ucdrBuffer reply_ub;
+
+        ucdr_init_buffer(&reply_ub, reply_buffer, sizeof(reply_buffer));
+        bool serialize_success = true;
+        serialize_success &= ucdr_serialize_bool(&reply_ub, status);
+        serialize_success &= ucdr_serialize_uint8_t(&reply_ub, curr_mode);
+        if (serialize_success == false || reply_ub.error) {
+            break;
+        }
+
+        request_sample_count++;
```

Comment:
what is this for?

Reply:
This is from the example. I think it is intended for use with the `uxr_run_session_until_all_status` version of the `uxr_run_session_*` functions, where you'd capture the `request_id` for each topic / replier when serialising into the buffer.